src/m365/spo/commands/tenant/tenant-applicationcustomizer-add.ts

```diff
@@ -0,0 +1,189 @@
+      {
+        option: '--webTemplate [webTemplate]'
+      }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        if (!validation.isValidGuid(args.options.clientSideComponentId)) {
+          return `${args.options.clientSideComponentId} is not a valid GUID`;
+        }
+
+        return true;
+      }
+    );
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    try {
+      if (this.verbose) {
+        logger.logToStderr(`Adding application customizer with clientSideComponentId ${args.options.clientSideComponentId} and title ${args.options.title} to the tenant app catalog.`);
+      }
+      const appCatalogUrl = await this.getAppCatalogUrl();
+      if (this.verbose) {
+        logger.logToStderr(`Got tenant app catalog url: ${appCatalogUrl}`);
+      }
+
+      const solutionId = await this.getSolutionIdFromComponentManifestList(appCatalogUrl, args.options.clientSideComponentId, logger);
+      const solution = await this.getSolutionFromAppCatalog(appCatalogUrl, solutionId, logger);
+
+      if (!solution.ContainsTenantWideExtension) {
+        throw `The solution does not contain an extension that can be deployed to all sites. Make sure that you've entered the correct component Id.`;
+      }
+      else if (!solution.SkipFeatureDeployment) {
+        throw 'The solution has not been deployed to all sites. Make sure to deploy this solution to all sites.';
+      }
+      await this.addTenantWideExtension(appCatalogUrl, args.options, logger);
+    }
+    catch (err: any) {
+      this.handleRejectedODataJsonPromise(err);
+    }
+  }
+
+  private async getAppCatalogUrl(): Promise<string> {
+    const spoTenantAppCatalogUrlGetCommandOutput: CommandOutput = await Cli.executeCommandWithOutput(spoTenantAppCatalogUrlGetCommand as Command, { options: { output: 'text', _: [] } });
+    const appCatalogUrl: string | undefined = spoTenantAppCatalogUrlGetCommandOutput.stdout;
+    if (!appCatalogUrl) {
+      throw 'Cannot add tenant-wide application customizer as app catalog cannot be found';
+    }
+    return appCatalogUrl;
+  }
+
+  private async getSolutionIdFromComponentManifestList(appCatalogUrl: string, clientSideComponentId: string, logger: Logger): Promise<string> {
+    if (this.verbose) {
+      logger.logToStderr('Retrieving component manifest item from the ComponentManifests list on the app catalog site so that we get the solution id');
+    }
+
+    const camlQuery = `<View><ViewFields><FieldRef Name='ClientComponentId'></FieldRef><FieldRef Name='SolutionId'></FieldRef></ViewFields><Query><Where><Eq><FieldRef Name='ClientComponentId' /><Value Type='Guid'>${clientSideComponentId}</Value></Eq></Where></Query></View>`;
+    const commandOptions: spoListItemListCommandOptions = {
+      webUrl: appCatalogUrl,
+      listUrl: `${urlUtil.getServerRelativeSiteUrl(appCatalogUrl)}/Lists/ComponentManifests`,
+      camlQuery: camlQuery,
```

Comment:
Could you add the fields option and determine what you want to retrieve. otherwise, you'll get all properties in the HTTP response, which is a waste of resources.

Reply:
I specify the fields to retrieve in the camlQuery, which makes the option `fields` redundant.